wremnants/muon_efficiencies.py

```diff
@@ -66,52 +66,90 @@ def make_muon_efficiency_helpers(filename = data_dir + "/testMuonSF/scaleFactorP
+        # same SFs for + and - for now
+        sf_tracking.view(flow=True)[:, :, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_tracking.view(flow=True)[:, :, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+
+        ## FIXME: avoid code duplication?
+        ## now again for reco
+        hist_name = f"fullSF2D_{nom_syst_tag}_reco_{eratag}"
+        hist_root = fin.Get(hist_name)
+        hist_hist = narf.root_to_hist(hist_root, axis_names = ["SF eta", "SF pt"])
 
+        if sf_reco is None:
+            axis_eta_eff_reco = hist_hist.axes[0]
+            axis_pt_eff_reco = hist_hist.axes[1]
+            sf_reco = hist.Hist(axis_eta_eff_reco, axis_pt_eff_reco, axis_charge, axis_nom_alt,  name = "sf_reco", storage = hist.storage.Weight())
 
         # same SFs for + and - for now
-        sf_tracking_reco.view(flow=True)[:, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
-        sf_tracking_reco.view(flow=True)[:, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_reco.view(flow=True)[:, :, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_reco.view(flow=True)[:, :, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
 
+        
     # set overflow and underflow equal to adjacent bins
-    sf_tracking_reco.view(flow=True)[0, ...] = sf_tracking_reco.view(flow=True)[1, ...]
-    sf_tracking_reco.view(flow=True)[axis_eta_eff_tracking_reco.extent-1, ...] = sf_tracking_reco.view(flow=True)[axis_eta_eff_tracking_reco.extent-2, ...]
+    sf_tracking.view(flow=True)[0, ...] = sf_tracking.view(flow=True)[1, ...]
+    sf_tracking.view(flow=True)[axis_eta_eff_tracking.extent-1, ...] = sf_tracking.view(flow=True)[axis_eta_eff_tracking.extent-2, ...]
+    sf_tracking.view(flow=True)[:, 0, ...] = sf_tracking.view(flow=True)[:, 1, ...]
+    sf_tracking.view(flow=True)[:, axis_pt_eff_tracking.extent-1, ...] = sf_tracking.view(flow=True)[:, axis_pt_eff_tracking.extent-2, ...]
+    # set overflow and underflow equal to adjacent bins
+    sf_reco.view(flow=True)[0, ...] = sf_reco.view(flow=True)[1, ...]
+    sf_reco.view(flow=True)[axis_eta_eff_reco.extent-1, ...] = sf_reco.view(flow=True)[axis_eta_eff_reco.extent-2, ...]
+    sf_reco.view(flow=True)[:, 0, ...] = sf_reco.view(flow=True)[:, 1, ...]
+    sf_reco.view(flow=True)[:, axis_pt_eff_reco.extent-1, ...] = sf_reco.view(flow=True)[:, axis_pt_eff_reco.extent-2, ...]
 
+    
     fin.Close()
 
     # TODO implement a convenient way to use enums for the bool/category/nom-var axes such that
     # the C++ code is less error-prone?
 
-    netabins = axis_eta_eff.size
+    netabins = axis_eta_eff.size  # this is the same also for reco and tracking, should one check explicitly?
```

Comment:
Can't hurt... Just check and raise and exception if not.

Reply:
Done